google/cloud/bigquery/job/base.py

```diff
@@ -73,6 +74,16 @@ def _error_result_to_exception(error_result):
     )
 
 
+ReservationUsage = namedtuple("ReservationUsage", "name slot_ms")
```

Comment:
Interesting!

This is a little different from our usual "class with `_properties` dictionary" approach, but does seem to save us some boilerplate. I'll allow it. :-)

Do you think we should do this for our other read-only classes?

Reply:
I though a namedtuple would better communicate that it's a read-only (output-only) property, although our "_properties classes"  can also have read-only properties - but yeah, a namedtuple requires much less boilerplate.

(FWIW, I personally view these output-only classes merely as named read-only containers with descriptive attribute names, and using a class with a bunch of read-only properties is just an implementation detail)

We can eventually refactor other output-only classes to namedtuples for consistency, yes, but I wouldn't do it just yet. There's not that much immediate value in it, as that code has already been written, and there might be some legit use cases out there that we don't know about... let's first wait and see how a namedtuple performs when released into the wild.